### PR TITLE
fix: fmt should check for unstaged files

### PR DESCRIPTION
**Pull request by mtojek**
Spotted in: https://github.com/coder/coder/actions/runs/3643167089/jobs/6151099869

```
--- shfmt
dogfood/main.tf
--- prettier
yarn run v1.22.19
$ prettier --cache --check '**/*.{css,html,js,json,jsx,md,ts,tsx,yaml,yml}' . ../ADOPTERS.md ../README.md ../docs
Checking formatting...
All matched files use Prettier code style!
Done in 2.8[6](https://github.com/coder/coder/actions/runs/3643167089/jobs/6151099869#step:6:7)s.
```

File `dogfood/main.tf` hasn't been formatted before, but `fmt` doesn't complain about it.

### CODE REVIEW

**Review by deansheather — APPROVED** [review]

File: .github/workflows/coder.yaml
Comment (mafredri) [inline .github/workflows/coder.yaml:278]: I thinks this is good, should we also consider doing a CI conditional for `terraform fmt -check` like we do for prettier and shfmt?

**Review by mafredri — APPROVED** [review]

File: .github/workflows/coder.yaml
Comment (mtojek) [reply .github/workflows/coder.yaml:278]: Sorry, I noticed your comment after merging. I can iterate on this if you like, but I assume that `check_unstaged.sh` as an ultimate check should work here well.

Comment (mafredri) [reply .github/workflows/coder.yaml:278]: No worries, I'll leave it up to you. Ultimately this check is great to have as a catch-all, the other is just a minor improvement which doesn't bring much benefit.

**Pull request merged**
